### PR TITLE
Changed Select and TextInput to accept a11yTitle

### PR DESCRIPTION
**Pull request by ShimiSun**
#### What does this PR do?
Propagates the a11yTitle prop as `aria-label` prop to TextInput, hence fixes the aria-label issue for both Select and TextInput components.
#### Where should the reviewer start?
TextInput.js
#### What testing has been done on this PR?
Added unit tests and verified snapshots are containing the `aria-label` prop.

#### How should this be manually tested?
snapshots
#### Any background context you want to provide?
a request for this fix came from @MikeKingdom
https://codesandbox.io/s/restless-smoke-g3zjc?file=/src/App.js
#### Do the grommet docs need to be updated?
no
#### Should this PR be mentioned in the release notes?
yes
#### Is this change backwards compatible or is it a breaking change?
backwards compatible

### CODE REVIEW

**Comment by MikeKingdom** [comment]
Thanks Shimi!

File: src/js/components/Select/__tests__/__snapshots__/Select-test.js.snap [hunk 1]
```diff
@@ -235,6 +235,241 @@ exports[`Select 0 value 1`] = `
...
+    className="c2"
+  >
+    <div
+      className="c3"
+    >
+      <div
+        className="c4"
+      >
+        <input
+          aria-label="aria-test, "
```
Comment (ericsoderberghp) [inline src/js/components/Select/__tests__/__snapshots__/Select-test.js.snap:433]: Why the ", " at the end?

Comment (ShimiSun) [reply src/js/components/Select/__tests__/__snapshots__/Select-test.js.snap:433]: The way that Select is processing and assigning `a11yTitle` is a little funky https://github.com/grommet/grommet/blob/d691e84055eee8a60f13dee99c97b7e311e81fb7/src/js/components/Select/Select.js#L246

I didn't dive into this existing logic (it's been there for a couple of years), but I'm guessing it is to allow concatenation of strings.

I can take a deeper look into it if needed.

Comment (ericsoderberghp) [reply src/js/components/Select/__tests__/__snapshots__/Select-test.js.snap:433]: In Select.js, I think we should make this change...
from:
```
a11yTitle={
  a11yTitle &&
  `${a11yTitle}${
    typeof value === 'string' ? `, ${value}` : ''
  }`
}
```
to:
```
a11yTitle={
  a11yTitle &&
  `${a11yTitle}${
    value && typeof value === 'string' ? `, ${value}` : ''
  }`
}
```

Comment (ShimiSun) [reply src/js/components/Select/__tests__/__snapshots__/Select-test.js.snap]: Done. the snapshot looks as expected now.

**Pull request merged**